packages/rainbowkit/src/components/RainbowKitProvider/AuthenticationContext.tsx

```diff
@@ -77,6 +82,24 @@ export function RainbowKitAuthenticationProvider<Message = unknown>({
     }
   }, [status, adapter, isDisconnected]);
 
+  const handleChangedAccount = ({ account }: ConnectorData) => {
+    // Only if account is changed then signOut
+    if (account) adapter.signOut();
+  };
+
+  // biome-ignore lint/nursery/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    if (connector && status === 'authenticated') {
+      // Attach the event listener when status is 'authenticated'
+      connector.on('change', handleChangedAccount);
+
```

Comment:
Also one thing here. We don't want to listen to `"change"` event unless is logged in with SIWE. If we listen to the event right after user connects his wallet then we can't log him out since he is not authenticated yet. 

Let me know if this ok approach 👍